apps/src/componentLibrary/slider/Slider.tsx

```diff
@@ -1,47 +1,96 @@
 import classnames from 'classnames';
-import React, {ChangeEvent, HTMLAttributes} from 'react';
+import React, {
+  ChangeEvent,
+  HTMLAttributes,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
 
-import {componentSizeToBodyTextSizeMap} from '@cdo/apps/componentLibrary/common/constants';
-import {ComponentSizeXSToL} from '@cdo/apps/componentLibrary/common/types';
-import Typography from '@cdo/apps/componentLibrary/typography';
+import {Button, ButtonProps} from '@cdo/apps/componentLibrary/button';
 
 import moduleStyles from './slider.module.scss';
 
 export interface SliderProps extends HTMLAttributes<HTMLInputElement> {
-  /** Slider onChange handler*/
+  /** Slider onChange handler
+   * @param {ChangeEvent<HTMLInputElement>} event - The change event triggered by the slider. */
   onChange: (event: ChangeEvent<HTMLInputElement>) => void;
   /** The name attribute specifies the name of an input element.
-     The name attribute is used to reference elements in a JavaScript,
-     or to reference form data after a form is submitted.
-     Note: Only form elements with a name attribute will have their values passed when submitting a form. */
+   * The name attribute is used to reference elements in JavaScript,
+   * or to reference form data after a form is submitted.
+   * Note: Only form elements with a name attribute will have their values passed when submitting a form. */
   name: string;
   /** The value attribute specifies the value of an input element. */
-  value?: string;
+  value: number | string;
   /** Slider label */
   label?: string;
   /** Is Slider disabled */
   disabled?: boolean;
-  /** Size of Slider */
-  size?: ComponentSizeXSToL;
+  /** Color of the slider track
+   * @default 'black'* */
+  color?: 'black' | 'brand' | 'white';
+  /** Is the slider centered
+   * @default false */
+  isCentered?: boolean;
+  /** Is the slider in percent mode
+   * @default false */
+  isPercentMode?: boolean;
+  /** Step value for the slider
+   * @default 1 */
+  step?: number;
+  /** [WIP] Array of step values for the slider */
+  steps?: number[];
+  /** Is the slider in right-to-left mode
+   * @default false */
+  isRtl?: boolean;
```

Comment:
General question about RTL, not blocking for this PR - do you have any thoughts about handling RTL at a global level i.e. via context or data tags, similar to theme? I know that several if not most components might just work out of the box in RTL due to html elements they're using, but are there others like this component that need to orient themselves differently based on RTL?